be/src/vec/sink/vmysql_result_writer.cpp

```diff
@@ -624,269 +625,8 @@ Status VMysqlResultWriter<is_binary_format>::append_block(Block& input_block) {
-                                                                            rows_buffer, col_const);
-            }
-            break;
-        }
-        case TYPE_ARRAY: {
-            // Currently all functions only support single-level nested arrays，
-            // so we use Array's child scale to represent the scale of nested type.
-            scale = _output_vexpr_ctxs[i]->root()->type().children[0].scale;
-            if (type_ptr->is_nullable()) {
-                auto& nested_type =
-                        assert_cast<const DataTypeNullable&>(*type_ptr).get_nested_type();
-                auto& sub_type = assert_cast<const DataTypeArray&>(*nested_type).get_nested_type();
-                status = _add_one_column<PrimitiveType::TYPE_ARRAY, true>(
-                        column_ptr, result, rows_buffer, col_const, scale, {sub_type});
-            } else {
-                auto& sub_type = assert_cast<const DataTypeArray&>(*type_ptr).get_nested_type();
-                status = _add_one_column<PrimitiveType::TYPE_ARRAY, false>(
-                        column_ptr, result, rows_buffer, col_const, scale, {sub_type});
-            }
-            break;
-        }
-        case TYPE_STRUCT: {
-            if (type_ptr->is_nullable()) {
-                auto& nested_type =
-                        assert_cast<const DataTypeNullable&>(*type_ptr).get_nested_type();
-                auto& sub_types = assert_cast<const DataTypeStruct&>(*nested_type).get_elements();
-                status = _add_one_column<PrimitiveType::TYPE_STRUCT, true>(
-                        column_ptr, result, rows_buffer, col_const, scale, sub_types);
-            } else {
-                auto& sub_types = assert_cast<const DataTypeStruct&>(*type_ptr).get_elements();
-                status = _add_one_column<PrimitiveType::TYPE_STRUCT, false>(
-                        column_ptr, result, rows_buffer, col_const, scale, sub_types);
-            }
-            break;
-        }
-        case TYPE_MAP: {
-            if (type_ptr->is_nullable()) {
-                auto& nested_type =
-                        assert_cast<const DataTypeNullable&>(*type_ptr).get_nested_type(); //for map
-                status = _add_one_column<PrimitiveType::TYPE_MAP, true>(
-                        column_ptr, result, rows_buffer, col_const, scale, {nested_type});
-            } else {
-                status = _add_one_column<PrimitiveType::TYPE_MAP, false>(
-                        column_ptr, result, rows_buffer, col_const, scale, {type_ptr});
-            }
-            break;
-        }
-        default: {
-            LOG(WARNING) << "can't convert this type to mysql type. type = "
-                         << _output_vexpr_ctxs[i]->root()->type();
-            return Status::InternalError("vec block pack mysql buffer failed.");
-        }
-        }
+        RETURN_IF_ERROR(type_ptr->get_serde()->write_column_to_mysql(*column_ptr, rows_buffer, 0, 0,
```

Comment:
get_serde will return temporary DataTypeSerDeSPtr and may lead to bad performace

Reply:
yes , later I would use create_data_type_serdes